facebook/draft-js
# Always use \ n for new line detection

**Summary**

Draft.js uses < br > for new line detection on all browsers except for IE. This can sometimes be problematic because most browsers[1] removes the < br > as user type in a new line. When react wants to remove the < br >, it may already be gone.

In https://github.com/facebook/draft-js/issues/1845 , a react exception is thrown when user tries to input with Windows 10 touch keyboard. This PR propose to always use \ n (rather than < br >) for new line detection, which fixes the issue. 

[1] Latest version of Edge, Chrome, Safari, but not Firefox. https://lan9.github.io/keyevent.html is a minimal test case --- use different browsers and focus on the second line, type something, noticed that the < br > is removed by browser.

**Test Plan**

Steps:
- Enable Windows 10 Touch Keyboard. This requires a windows device with keyboard detached (tablet mode). It's possible to simulate this environment with VMWare on Mac if you tweak the config to simulate touch support, and force enable tablet mode in settings.
- Use Edge to open draftjs.org, swipe hello on the Touch keyboard.
Observe error in console. At this moment editorstate can be in a bad state. (can't type / delete / error in console).
This is a demo of the steps: https://twitter.com/DevYuze/status/1034207906206429184

Follow the same step to verify the fix on: https://jsfiddle.net/lang1z/ygmzjf31/14/


**Discussion**
1) If most browsers remove the < br >, why is it only a problem on Edge with Touch keyboard?
There are two hypothesis, but they are not verified:
Hypothesis 1: because both browser and react try to remove the < br >, it's a race condition. If React remove it before browser does, it's fine. Otherwise it's not.
Hypothesis 2: React's maker knows some browsers remove the < br >, so it has logic to adapt to it. Because Touch keyboard triggers wrong key events (you can test it on https://jsfiddle.net/lang1z/ygmzjf31/14/), React may not be able to adapt.

2) Why did the code base used  < br > in the first place?
It's very obvious the original author deliberately choose to use < br > on most browsers. The motivation is unknown to me. I tested  https://jsfiddle.net/lang1z/ygmzjf31/14/ on reasonably old browser: Chrome 55, Firefox 45 and Safari 10, new line rendering and typing all seem fine.
An safer alternative is to only switch Edge to use \ n, as the issue really only surfaced on Edge, like what we've done in twitter-forked draft-js: https://github.com/twitter-forks/draft-js/pull/8. However, I feel such patches may not be healthy for the codebase in long term.
